conn.go

```diff
@@ -358,6 +359,15 @@ func (hc *halfConn) decrypt(b *block) (ok bool, prefixLen int, alertValue alert)
 				hc.additionalData[11] = byte(n >> 8)
 				hc.additionalData[12] = byte(n)
 				additionalData = hc.additionalData[:]
+			} else {
+				if len(payload) > int(uint16(1<<14)+256) {
+					return false, 0, alertRecordOverflow
+				}
+				// Check AD header, see 5.2 of RFC8446
+				additionalData = make([]byte, 5)
+				additionalData[0] = byte(recordTypeApplicationData)
+				binary.BigEndian.PutUint16(additionalData[1:], VersionTLS12)
```

Comment:
In other places in the code 16-bit values are explicitly split using bitmasks. (E.g.: `hc.additionalData[11] = byte(payloadLen >> 8)`, `hc.additionalData[12] = byte(payloadlen)`.) While I think that using `encoding/binary` is more clear to the reader, it makes it inconsistent within this file.